main_test.go

```diff
@@ -18,7 +19,7 @@ type ComponentTest struct {
 }
 
 func (f *ComponentTest) InitializeScenario(ctx *godog.ScenarioContext) {
-	component := steps.NewComponent()
+	component := steps.NewComponent(context.Background())
```

Comment:
Should we not use the context passed in to the function rather than generating a new one?
```suggestion
	component := steps.NewComponent(ctx)
```

Reply:
If I use the *godog.ScenarioContext as the context to pass in there are lots of error messages, as the dp-kafka library needs the context.Background() type to work.  I've changed where the context is set for the test now so hopefully it will be a bit tidier.